src/input/composite_device/mod.rs

```diff
@@ -1736,9 +1736,36 @@ impl CompositeDevice {
             return Ok(());
         }
 
-        // Stop all old target devices
-        let targets_to_stop = self.target_devices.clone();
-        for (path, target) in targets_to_stop.into_iter() {
+        // Identify which target devices are new
+        let mut device_types_to_start: Vec<String> = vec![];
+        for kind in device_types.clone() {
+            if self.target_kind_running(kind.clone()).await? {
```

Comment:
We can just iterate on this instead of using `clone()`.

e.g.
```rust
for kind in device_types.iter() {
    if self.target_kind_running(kind.as_str()).await? {
       ...
   ...
}
```